File: Java Spring Boot Project/misc/nport_project/MtCProject1/src/main/java/com/nikosportolos/MtCProject1/models/responses/DepartmentResponse.java
package com.nikosportolos.MtCProject1.models.responses;

import com.nikosportolos.MtCProject1.models.BusinessUnit;

public class DepartmentResponse {

    /**
     * Instance variables
     **/

    private long id;
    private String name;
    private BusinessUnit businessUnit;


    /**
     * Constructors
     **/

    public DepartmentResponse() {
    }

    public DepartmentResponse(long id, String name, BusinessUnit businessUnit) {
        this.id = id;
        this.name = name;
        this.businessUnit = businessUnit;
    }

    /**
     * Getters / Setters
     **/

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public BusinessUnit getBusinessUnit() {
        return businessUnit;
    }

    public void setBusinessUnit(BusinessUnit businessUnit) {
        this.businessUnit = businessUnit;
    }
}

File: Java Spring Boot Project/company_directory_app/lib/models/responses/department_response.dart
import 'package:company_directory_app/models/responses/business_unit_response.dart';

class DepartmentResponse {
  final int id;
  final String name;
  final BusinessUnitResponse businessUnit;

  DepartmentResponse({this.id, this.name, this.businessUnit});

  factory DepartmentResponse.fromJson(Map<String, dynamic> json) {
    return DepartmentResponse(id: json['id'], name: json['name'], businessUnit: BusinessUnitResponse.fromJson(json['businessUnit']));
  }
}
